### PR TITLE
Support full media queries in `theme.breakpoints` in @theme-ui/match-media hooks

### LINKED ISSUE #2197: `useBreakpointIndex` and `useResponsiveValue` don't handle full `@media` queries in `theme.breakpoints`
`useBreakpointIndex` and `useResponsiveValue` don't handle full `@media` queries in `theme.breakpoints`.

Thanks for finding this @danimadmolil 🙏.

### Discussed in https://github.com/system-ui/theme-ui/discussions/2195

<div type='discussions-op-text'>

<sup>Originally posted by **danimadmolil** April  7, 2022</sup>
Hellow, i'm new to theme-ui. my project is simple react-app.on index.js i create a theme and use it with ThemeProvider like this:
### index.js
```
const theme = {
 breakpoints: [
    "@media screen and (min-width:300px) and (max-width:499px)",
    "@media screen and (min-width: 500px) and (max-width:899px)",
    "@media screen and (min-width: 900px)",
  ],
} 
<ThemeProvider theme={theme}>
   <App/>
</ThemeProvider>

```
### App.js
```
import { useResponsiveValue, useBreakpointIndex } from "@theme-ui/match-media";
function App(){
  const color = useResponsiveValue((theme) => ["red", "blue", "orange"]);
  return <div className="app" style={{backgroundColor:color}}></div>
} 
```

the result color is always red. i'm not whay it's like that. can any one help me please?
**note**:if i use breakpoint:['40em','50em','60em'] it work fine but i like to use media queries with condition like (min-width) and (max-width).</div>

### PR DESCRIPTION
**Pull request by hasparus**
Closes #2197.

So, turns out `@theme-ui/match-media` isn't too popular and we just didn't remember about it when we added full media query support to responsive values in `sx` prop.

<details>
  <summary>📦 Published PR as canary version: <code>0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @theme-ui/color-modes@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/color@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/components@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/core@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/css@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/custom-properties@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/editor@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install gatsby-plugin-theme-ui@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install gatsby-theme-style-guide@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install gatsby-theme-ui-layout@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/match-media@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/mdx@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/parse-props@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-base@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-bootstrap@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-bulma@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-dark@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-deep@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-funk@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-future@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-polaris@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-roboto@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-sketchy@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-swiss@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-system@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-tailwind@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/preset-tosh@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/presets@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/prism@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/sidenav@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/style-guide@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/tailwind@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/theme-provider@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install theme-ui@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  npm install @theme-ui/typography@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  # or 
  yarn add @theme-ui/color-modes@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/color@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/components@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/core@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/css@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/custom-properties@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/editor@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add gatsby-plugin-theme-ui@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add gatsby-theme-style-guide@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add gatsby-theme-ui-layout@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/match-media@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/mdx@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/parse-props@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-base@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-bootstrap@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-bulma@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-dark@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-deep@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-funk@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-future@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-polaris@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-roboto@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-sketchy@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-swiss@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-system@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-tailwind@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/preset-tosh@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/presets@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/prism@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/sidenav@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/style-guide@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/tailwind@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/theme-provider@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add theme-ui@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  yarn add @theme-ui/typography@0.14.3--canary.2199.c124ac7c4eaa771a50ae43f2dd1bc0aa7ca2bdab.0
  ```
</details>

### CODE REVIEW

File: packages/match-media/src/index.ts [hunk 1]
```diff
@@ -25,9 +25,13 @@ export const useBreakpointIndex = (options: defaultOptions = {}) => {
   const [value, setValue] = useState(defaultIndex)
   useEffect(() => {
     const getIndex = () =>
-      breakpoints.filter(
-        (bp) => window.matchMedia(`screen and (min-width: ${bp})`).matches
-      ).length
+      breakpoints.filter((bp) => {
+        const query = bp.includes('@media')
+          ? bp.slice(7) // remove '@media '
```
Comment (lachlanjc) [inline packages/match-media/src/index.ts]: `bp.replace('@media ', '')` might be more readable, I don't feel strongly though

**Review by lachlanjc — APPROVED** [review]
Source looks good to me!

File: packages/match-media/src/index.ts (hunk 1 above)
Comment (hasparus) [reply packages/match-media/src/index.ts]: Fair point 🙏 My code is actually buggy. If you have a leading space (like `  @media `), my version wouldn't work, and yours is more permissive. Applying your suggestion!

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.14.3`](https://github.com/system-ui/theme-ui/releases/tag/v0.14.3) :rocket:

**Pull request merged**